File: standard/src/main/java/com/KillerBLS/modpeide/manager/interfaces/OnPanelClickListener.java
/*
 * Copyright (C) 2018 Light Team Software
 *
 * This file is part of ModPE IDE.
 *
 * ModPE IDE is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * ModPE IDE is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see
 * <http://www.gnu.org/licenses/>.
 */

package com.KillerBLS.modpeide.manager.interfaces;

/**
 * Обработка нажатий меню Toolbar'а.
 */
public interface OnPanelClickListener {
    void onNewButton();
    void onOpenButton();
    void onSaveButton();
    void onPropertiesButton();
    void onCloseButton();
    void onCutButton();
    void onCopyButton();
    void onPasteButton();
    void onSelectAllButton();
    void onSelectLineButton();
    void onDeleteLineButton();
    void onDuplicateLineButton();
    void onFindButton();
    void onReplaceAllButton();
    void onGoToLineButton();
    void onSyntaxValidatorButton();
    void onInsertColorButton();
    //void onDownloadSourceButton();
    void onUndoButton();
    void onRedoButton();
    void onSettingsButton();
}

File: presentation/src/main/kotlin/com/lightteam/modpeide/presentation/main/activities/interfaces/OnPanelClickListener.kt
/*
 * Licensed to the Light Team Software (Light Team) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The Light Team licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.lightteam.modpeide.presentation.main.activities.interfaces

interface OnPanelClickListener {
    fun onDrawerButton()

    fun onNewButton()
    fun onOpenButton()
    fun onSaveButton()
    fun onPropertiesButton()
    fun onCloseButton()

    fun onCutButton()
    fun onCopyButton()
    fun onPasteButton()
    fun onSelectAllButton()
    fun onSelectLineButton()
    fun onDeleteLineButton()
    fun onDuplicateLineButton()

    fun onFindButton()
    fun onReplaceAllButton()
    fun onGoToLineButton()

    fun onSyntaxValidatorButton()
    fun onInsertColorButton()
    //fun onDownloadSourceButton()

    fun onUndoButton()
    fun onRedoButton()

    fun onSettingsButton()
}